### PR TITLE
Update JAX to use new math libraries in ROCm-5.0.

### PR DESCRIPTION
**Pull request by reza-amd**
This PR integrates new math libraries added in ROCm-5.0 in JAX.
A few unit tests have been marked to be skipped but they should be fixed in next ROCm release.

### CODE REVIEW

**Review by hawkinsp — COMMENTED** [review]
Thanks for the PR!

I'm happy to see the CI/docker stuff: that promises to be useful if we ever need to debug a ROCm problem.

File: build/rocm/run_single_gpu.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
```
Comment (hawkinsp) [inline build/rocm/run_single_gpu.py:1]: All .py files have to have the apache license header. (Possibly other files also, but certainly the .py files.)

Same everywhere.

File: build/rocm/ci_build.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (reza-amd) [reply build/rocm/ci_build.sh:1]: Done.

File: build/rocm/run_single_gpu.py (hunk 1 above)
Comment (reza-amd) [reply build/rocm/run_single_gpu.py:1]: Done.

**Pull request merged**